### PR TITLE
Deduplicate lab 13

### PR DESCRIPTION
**Pull request by pavpanchekha**
This PR uses `wbetools.patch` to remove duplicate definitions in `lab13.py`. This doesn't affect the book. There are still a non-trivial number of duplicate methods because this chapter adds superclasses to the `DrawXXX` methods. We can't patch in a superclass (it's possible in Python, but I don't think it's possible in JavaScript) so we have to copy and paste some code. But a lot less than is in there now!

The benefit of deduplicating is that code won't get desynchronized. For example, I noticed while doing this that our changes to `previous_word` in Chapter 7 hadn't made it into Chapter 13.

### CODE REVIEW

**Comment by pavpanchekha** [comment]
I've confirmed that the compiled `lab13.js` still works great.

File: src/lab13.py [hunk 1]
```diff
@@ -354,120 +355,12 @@ def body(self):
...
-        self.cursor_x += w + font.measureText(" ")
-
-    def input(self, node):
-        w = INPUT_WIDTH_PX
-        if self.cursor_x + w > self.width:
-            self.new_line()
-        line = self.children[-1]
-        input = InputLayout(node, line, self.previous_word)
-        line.children.append(input)
-        self.previous_word = input
```
Comment (chrishtr) [inline src/lab13.py:448]: self.previous_word is not defined in lab11

**Review by chrishtr — COMMENTED** [review]
Is it easy to print out the fully inlined versions and diff them? Worried I missed other things.

File: src/lab13.py [hunk 2]
```diff
@@ -1045,8 +776,6 @@ def render(self):
             self.needs_paint = False
 
         clamped_scroll = self.clamp_scroll(self.scroll)
-        if clamped_scroll != self.scroll:
```
Comment (chrishtr) [inline src/lab13.py:1048]: Why is this no longer needed?

File: src/lab13.py (hunk 1 above)
Comment (pavpanchekha) [reply src/lab13.py:448]: Correct—this was a simplification we failed to propagate to lab13.

File: src/lab13.py (hunk 2 above)
Comment (pavpanchekha) [reply src/lab13.py:1048]: If you look a line or two down, this `if` statement is duplicated. It doesn't have any effects we need to repeat twice, either. Probably a copy-and-paste bug.

**Comment by pavpanchekha** [comment]
I went through a diff. This PR fixes two more tiny bugs:
- in `BlockLayout.word`, current main calls `get_font(size, weight, size)`, meaning that we always measure roman text even if it's supposed to be italics.
- The space after input areas didn't do pt to px conversion so was too big.
Plus it pulls in a small refactor to `layout_mode` and some changes to dead code / record calls. Basically, I checked the diff and the diff is good.

**Pull request merged**